contracts/DataFlat.sol

```diff
@@ -454,6 +454,17 @@ contract SwapData is AccessControl {
         return listings;
     }
 
+    function readListingsByIndex(
+        uint256 start,
+        uint256 end
+    ) external view returns (SwapListing[] memory) {
+        SwapListing[] memory listings = new SwapListing[](end - start + 1);
+        for (uint256 i = start; i <= end; i++) {
```

Comment:
The loop start to end only required when _listingIdTracker.current()-1 is greater then end. or else start to _listingIdTracker.current() - 1 to prevent empty places in the array returned by the contract